sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -0,0 +1,187 @@
+       */
+      EXECUTE_BUNDLE,
+      /**
+       * Create Method creates a single FHIR resource @see <a
+       * href=https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/create></a>.
+       */
+      CREATE
+    }
+
+    /**
+     * Gets Fhir store.
+     *
+     * @return the Fhir store
+     */
+    abstract String getFhirStore();
+
+    /**
+     * Gets write method.
+     *
+     * @return the write method
+     */
+    abstract WriteMethod getWriteMethod();
+
+    /** The type Builder. */
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      /**
+       * Sets Fhir store.
+       *
+       * @param fhirStore the Fhir store
+       * @return the Fhir store
+       */
+      abstract Builder setFhirStore(String fhirStore);
+
+      /**
+       * Sets write method.
+       *
+       * @param writeMethod the write method
+       * @return the write method
+       */
+      abstract Builder setWriteMethod(WriteMethod writeMethod);
+
+      /**
+       * Build write.
+       *
+       * @return the write
+       */
+      abstract Write build();
+    }
+
+    private static Write.Builder write(String fhirStore) {
+      return new AutoValue_FhirIO_Write.Builder().setFhirStore(fhirStore);
+    }
+
+    /**
+     * Create Method creates a single FHIR resource. @see <a
+     * href=https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/create></a>
+     *
+     * @param fhirStore the hl 7 v 2 store
+     * @return the write
+     */
+    public static Write create(String fhirStore) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore)
+          .setWriteMethod(Write.WriteMethod.CREATE)
+          .build();
+    }
+
+    /**
+     * Execute Bundle Method executes a batch of requests as a single transaction @see <a
+     * href=https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/executeBundle></a>.
+     *
+     * @param fhirStore the hl 7 v 2 store
+     * @return the write
+     */
+    public static Write executeBundles(String fhirStore) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore)
+          .setWriteMethod(WriteMethod.EXECUTE_BUNDLE)
+          .build();
+    }
+
+    @Override
+    public PDone expand(PCollection<HttpBody> messages) {
+      messages.apply(ParDo.of(new WriteFhirFn(this.getFhirStore(), this.getWriteMethod())));
+      return PDone.in(messages.getPipeline());
+    }
+  }
+
+  /** The type Write Fhir fn. */
+  static class WriteFhirFn extends DoFn<HttpBody, PDone> {
+    // TODO when the healthcare API releases a bulk import method this should use that to improve
+    // throughput.
+
+    private transient HealthcareApiClient client;
```

Comment:
You might want to reconstruct the transient objects during deserialization, otherwise you will probably get NPEs.

Reply:
The lifecycle of a DoFn (IMHO described best in this [SO Post](https://stackoverflow.com/questions/45985753/what-is-the-difference-between-dofn-setup-and-dofn-startbundle)) tells us that a JVM process responsible for running this DoFn, the `@Setup` annotated method initClient will always be called before the the `@ProcessElement` annotated method.

I will look out for NPEs here as I write the integration tests.